app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -4101,12 +4161,55 @@ class ChatActivity :
         if (file.exists()) {
             share(message)
         } else {
-            downloadFileToCache(message, false) {
+            downloadFileToCache(message,  false ) {
                 share(message)
             }
         }
     }
 
+
+    fun saveImage(message: ChatMessage){
+        if (permissionUtil.isFilesPermissionGranted()) {
+            saveImageToStorage(message ,  false){
+            }
+        }
+        else{
+             UploadAndShareFilesWorker.requestStoragePermission(this@ChatActivity)
+        }
+    }
+
+     fun showWarningDialog(message: ChatMessage) {
```

Comment:
maybe better find a name that tells a bit more about what this dialog does. For example "showSaveToStorageWarning"